### ISSUE TITLE
Warning: openssl_decrypt(): IV passed is only 5 bytes long

### ISSUE THREAD

**Issue by eArabic** [halojoy/QuickCommunity#2]
Hello dear friend
I have the problem again What is the solution?
Warning: openssl_decrypt(): IV passed is only 5 bytes long, cipher expects an IV of precisely 16 bytes, padding with \0 in /membri/host/web/QuickCommunity-2.01/core/classSession.php on line 31
Warning: openssl_encrypt(): IV passed is only 5 bytes long, cipher expects an IV of precisely 16 bytes, padding with \0 in /membri/host/web/QuickCommunity-2.01/core/classSession.php on line 116

**Comment by halojoy** [comment]
The solution is to download the latest 'release'.
I have done a lot of changes and changed the handling of decrypt/encrypt.
If you do this you will have not this problem.
The latest is **version 2.02**.
Or download the 'trunk'.

**Comment by eArabic** [comment]
I have the problem again What is the solution?
![zx](https://user-images.githubusercontent.com/24383546/35775923-e3596058-09a3-11e8-92f2-c4fe6d5ade6a.png)

**Comment by halojoy** [comment]
This problem, I think, comes from file permission.
Setup needs to write to files. Write the IV for decrypt.

If 'classSession.php' have not write permission the script can not write to it.

Also the folders 'conf' needs to be writeable.
Also if you use SQLite setup, folder 'data' needs write permission.

But as I told you,
Download the latest version. It has many improvements.

**Comment by halojoy** [comment]
I see you use latest version.
This is a mystery.

My own install shows nothing wrong.
It is working perfect when I run it.

**Comment by halojoy** [comment]
Here is a solution you can try.

Change classSession.php line 24 to:
**$this->iv         = base64_decode($settings->cryptoiv);**

I found this on the web. I am not sure it works.

**Comment by halojoy** [comment]
Can you post the value you get in database of table 'settings':
cryptokey = ?
cryptoiv  = ?

**Comment by eArabic** [comment]
Hello
When you choose to install MYSQL
I have errors
**And when you select SQLITE
The previous errors do not appear after using the last update 2.04, so excellent.**
What is the MYSQL solution?

```
INSTALL step 3
Fatal error: Uncaught exception 'PDOException' with message 'SQLSTATE[21S01]: Insert value list does not match column list: 1136 Column count doesn't match value count at row 1' in /membri/user_1426587/Root/Quick/setup/setup3.php:73 Stack trace: #0 /membri/user_1426587/Root/Quick/setup/setup3.php(73): PDO->exec('INSERT INTO set...') #1 /membri/user_1426587/Root/Quick/setup.php(16): require('/membri/albasha...') #2 {main} thrown in /membri/user_1426587/Root/Quick/setup/setup3.php on line 73
```


INSTALL step 3
Fatal error: Uncaught exception 'PDOException' with message 'SQLSTATE[21S01]: Insert value list does not match column list: 1136 Column count doesn't match value count at row 1' in /membri/user_1426587/Root/Quick/setup/setup3.php:73 Stack trace: #0 /membri/user_1426587/Root/Quick/setup/setup3.php(73): PDO->exec('INSERT INTO set...') #1 /membri/user_1426587/Root/Quick/setup.php(16): require('/membri/albasha...') #2 {main} thrown in /membri/user_1426587/Root/Quick/setup/setup3.php on line 73

**Comment by eArabic** [comment]
Please add these features:
mobile browsing.
Editing css is easy, I noticed that I'm fairest and the edit does not appear.
The settings should limit the number of  posts that will appear on the page.
Features:
  - Search
  - Pagination
  - Locked and sticky posts
Link feature does not work It is one for all posts:
/?act=post&pid=**$n**  |  **$n** : 1,2,3,4,5,.....  
There should be a code that distinguishes the  post  from other  posts.
When a Post Reply to an article should be returned to the  post  page automatically.

**Comment by halojoy** [comment]
I have done a fix for the Fatal error in setup3.php Line 73.
Released it as 2.05

Can you please open new issue if you have other things to tell.
This issue was for the IV issue.
Thank You for making the code better.

**Comment by eArabic** [comment]
Thank you so much for taking the test, it works great without problems
